apps/omg_child_chain/test/omg_child_chain/block_queue/core_test.exs

```diff
@@ -340,164 +348,218 @@ defmodule OMG.ChildChain.BlockQueue.CoreTest do
+      assert [%{gas_price: 20_000_000_000}] =
+               empty
+               |> Core.enqueue_block("1", 1000, 0)
+               |> Core.get_blocks_to_submit()
     end
 
-    test "Calling with current ethereum height doesn't change the gas params", %{
-      empty_with_gas_params: empty_with_gas_params
-    } do
-      state = empty_with_gas_params
-
-      current_height = state.parent_height
-      current_price = state.gas_price_to_use
-      current_params = state.gas_price_adj_params
-
-      {_, newstate} = Core.set_ethereum_status(state, 1, 0, false)
-
-      assert current_height == newstate.parent_height
-      assert current_price == newstate.gas_price_to_use
-      assert current_params == newstate.gas_price_adj_params
+    test "Fresh block to submit gets the initial gas price, even if ethereum progressed before",
+         %{empty_high_max_gas_price: empty} do
+      assert [%{gas_price: 20_000_000_000}] =
+               empty
+               |> Core.set_ethereum_status(10, 0, false)
+               |> elem(1)
+               |> Core.enqueue_block("1", 1000, 10)
+               |> Core.get_blocks_to_submit()
     end
 
-    test "Gas price is lowered when ethereum blocks gap isn't filled", %{empty_with_gas_params: empty_with_gas_params} do
-      state = Core.enqueue_block(empty_with_gas_params, <<0>>, 6000, 1)
-      current_price = state.gas_price_to_use
-
-      {_, newstate} = Core.set_ethereum_status(state, 2, 0, false)
-
-      assert current_price > newstate.gas_price_to_use
-
-      # assert the actual gas price based on parameters value - test could fail if params or calculation will change
-      assert 90 == newstate.gas_price_to_use
+    test "Blocks enqueued in sequence both get the initial gas price", %{empty_high_max_gas_price: empty} do
+      assert [%{gas_price: 20_000_000_000}, %{gas_price: 20_000_000_000}] =
+               empty
+               |> Core.set_ethereum_status(0, 0, false)
+               |> elem(1)
+               |> Core.enqueue_block("1", 1000, 0)
+               |> Core.enqueue_block("2", 2000, 0)
+               |> Core.get_blocks_to_submit()
     end
 
-    test "Gas price is raised when ethereum blocks gap is filled", %{empty_with_gas_params: empty_with_gas_params} do
-      state = empty_with_gas_params
-      current_price = state.gas_price_to_use
-      eth_gap = state.gas_price_adj_params.eth_gap_without_child_blocks
-
-      {:do_form_block, newstate} =
-        state
-        |> Core.enqueue_block(<<0>>, 6000, 1)
-        |> Core.set_ethereum_status(1 + eth_gap, 0, false)
-
-      assert current_price < newstate.gas_price_to_use
-
-      # assert the actual gas price based on parameters value - test could fail if params or calculation will change
-      assert 200 == newstate.gas_price_to_use
+    test "Progressing to enqueue height may raise price if threshold exceeded", %{empty_high_max_gas_price: empty} do
```

Comment:
I had a hard time to understand test description, especially `may` & `if`
I'd like sth with more assurance ~ `exceeding the threshold height results with price raise`

Reply:
um, so the test and description are as un-assuring as the behavior. It means that despite the `set_ethereum_status` goes to exactly `enqueue_block`'s eth_height, it still raises the price.

I'll clarify to

```
    # NOTE: This behavior is strange - revisit along with the other ones (see `NOTEs` here)
    #       Why would detecting the exact height of enqueue cause the price to raise - we don't know if there's need to
    test "Despite progressing to enqueue height price may raise", %{empty_high_max_gas_price: empty} do
```